hera_pspec/pspecdata.py

```diff
@@ -376,129 +413,117 @@ def get_F(self, key1, key2, use_identity=False, true_fisher=False,
-            Use the identity matrix to weight the data, instead of the 
-            covariance matrix. Default: False.
-        
-        true_fisher : bool, optional
-            Whether to calculate the "true" Fisher matrix, or the "effective" 
-            matrix s.t. W=MF and p=Mq. Default: False. (FIXME)
-        
-        taper : str, optional
-            Tapering (window) function used when calculating Q. Takes the same 
-            arguments as aipy.dsp.gen_window(). Default: 'none'.
-        
         Returns
         -------
-        F : array_like, complex
+        G : array_like, complex
             Fisher matrix, with dimensions (Nfreqs, Nfreqs).
         """
         assert isinstance(key1, tuple)
         assert isinstance(key2, tuple)
-        F = np.zeros((self.Nfreqs, self.Nfreqs), dtype=np.complex)
-        
-        # Whether to use look-up fn. for identity or inverse covariance matrix
-        icov_fn = self.I if use_identity else self.iC
-        
-        # FIXME: k1,2 aren't allowed to be lists any more
-        #iC1, iC2 = 0, 0
-        #for _k in k1: iC1 += icov_fn(_k)
-        #for _k in k2: iC2 += icov_fn(_k)
-        iC1 = icov_fn(key1)
-        iC2 = icov_fn(key2)
-        
-        # Multiply terms to get the true or effective Fisher matrix
-        # FIXME: I think effective <=> true have been mixed up here
-        if true_fisher:
-            # This is for the "true" Fisher matrix
-            # FIXME: What is this for?
-            CE1, CE2 = {}, {}
-            Cemp1, Cemp2 = self.I(key1), self.I(key2)
-            
-            for ch in xrange(self.Nfreqs):
-                Q = self.get_Q(ch, self.Nfreqs, taper=taper)
-                # C1 Cbar1^-1 Q Cbar2^-1; C2 Cbar2^-1 Q Cbar1^-1
-                CE1[ch] = np.dot(Cemp1, np.dot(iC1, np.dot(Q, iC2)))
-                CE2[ch] = np.dot(Cemp2, np.dot(iC2, np.dot(Q, iC1)))
-            
-            for i in xrange(self.Nfreqs):
-                for j in xrange(self.Nfreqs):
-                    F[i,j] += np.einsum('ij,ji', CE1[i], CE2[j]) # C E C E
-        else:
-            # This is for the "effective" matrix s.t. W=MF and p=Mq
-            iCQ1, iCQ2 = {}, {}
-            
-            for ch in xrange(self.Nfreqs): # this loop is nchan^3
-                Q = self.get_Q(ch, self.Nfreqs, taper=taper)
-                iCQ1[ch] = np.dot(iC1, Q) #C^-1 Q
-                iCQ2[ch] = np.dot(iC2, Q) #C^-1 Q
-            
-            for i in xrange(self.Nfreqs): # this loop goes as nchan^4
-                for j in xrange(self.Nfreqs):
-                    F[i,j] += np.einsum('ij,ji', iCQ1[i], iCQ2[j]) #C^-1 Q C^-1 Q 
-        return F
-    
-    def get_MW(self, F, mode='F^-1'):
+        G = np.zeros((self.Nfreqs, self.Nfreqs), dtype=np.complex)
+
+        R1 = self._R(key1)
+        R2 = self._R(key2)
+        
+        iR1Q, iR2Q = {}, {}
+        for ch in xrange(self.Nfreqs): # this loop is nchan^3
+            iR1Q[ch] = np.dot(R1, Q) # R_1 Q
```

Comment:
I don't see where `Q` has been set in this function.

Reply:
Fixed; thank you!